app/src/main/java/io/appium/settings/Settings.java

```diff
@@ -70,12 +93,189 @@ public void onCreate(Bundle savedInstanceState) {
             LocationTracker.getInstance().start(this);
         }
 
+        handleRecording(getIntent());
+    }
+
+    private void handleRecording(Intent intent) {
+        if (intent == null) {
+            Log.e(TAG, "handleRecording: intent is null");
+            finishActivity();
+            return;
+        }
+
+        String recordingAction = intent.getAction();
+        if (recordingAction == null) {
+            Log.e(TAG, "handleRecording: intent.getAction() is null");
+            finishActivity();
+            return;
+        }
+
+        if (isLowerThanQ()) {
+            Log.e(TAG, "handleRecording: Current Android OS Version is lower than Q");
+            finishActivity();
+            return;
+        }
+
+        if (!areRecordingPermissionsGranted()) {
+            Log.e(TAG, "handleRecording: Required Permissions are not granted");
+            finishActivity();
+            return;
+        }
+
+        String recordingFilename = intent.getStringExtra(ACTION_RECORDING_FILENAME);
+
+        if (isValidFileName(recordingFilename)) {
+            String timeStamp = new SimpleDateFormat(
+                    "yyyy-MM-dd_HH-mm-ss", Locale.US).format(new Date());
+            recordingFilename = RecorderConstant.DEFAULT_RECORDING_FILENAME +
+                    "_" + timeStamp + ".mp4";
+            Log.w(TAG, "handleRecording: Invalid filename passed by user," +
+                    " using default one: " + recordingFilename);
+        }
+
+        recordingOutputPath = getExternalFilesDir(null).getAbsolutePath()
+                + File.separator + recordingFilename;
+
+        Display display = ((WindowManager) getSystemService(WINDOW_SERVICE)).getDefaultDisplay();
+
+        if (display != null) {
+            int deviceRotation = display.getRotation();
+
+            switch(deviceRotation) {
```

Comment:
do we support rotation change in runtime?

Reply:
not yet but it doesn't crash it as well if user change rotation of device in the middle of recording 👍